scripts/postgresql-setup.sh

```diff
@@ -185,24 +187,25 @@ function restore_snapshot {
 	if test "${file_count}" -gt 0 ; then
 	  echo "Ledger state directory ($2) is not empty. Please empty it and then retry."
 	  exit 1
-	  fi
+	fi
 	tmp_dir=$(mktemp "${directory}" -t db-sync-snapshot-XXXXXXXXXX)
 	tar xvf "$1" --directory "$tmp_dir"
 	if test -d "${tmp_dir}/db/" ; then
 	  # New pg_dump format
-	  lstate_gz_file=$(find "${tmp_dir}/" -iname "*.lstate.gz")
-	  lstate_file=$(basename "${lstate_gz_file}" | sed 's/.gz$//')
-	  gunzip --to-stdout "${lstate_gz_file}" > "$2/${lstate_file}"
+	  if [ -n "${lstate_gz_file:-}" ] ; then
+	    lstate_file=$(basename "${lstate_gz_file}" | sed 's/.gz$//')
+	    gunzip --to-stdout "${lstate_gz_file}" > "$2/${lstate_file}"
+	  fi
 
-          # Important: specify --schema=public below to skip over `create schema public`
-          # statement generated by pg_dump
+	  # Important: specify --schema=public below to skip over `create schema public`
+	  # statement generated by pg_dump
 	  pg_restore \
 	    --schema=public \
 	    --jobs="${numcores}" \
 	    --format=directory \
 	    --dbname="${PGDATABASE}" \
-            --no-owner \
-	    --exit-on-error \
+	    --no-owner \
+	    "$( [ -z "${SKIP_RESTORE_ERROR}" ] && echo " --exit-on-error" )" \
```

Comment:
Won't quoting this make it add `""` to the list of arguments? I suspect this will result in:

```
pg_restore: error: too many command-line arguments (first is "")
```